Assets/Scripts/Actors/ActorComponents/PlayerActorInventory.cs

```diff
@@ -88,29 +111,26 @@ public void CheckUseItem( RaycastHit hitInfo )
 
 	void CheckGiveResources( RaycastHit hitInfo )
 	{
-		if ( hitInfo.transform )
-		{
-			BuddyStats buddyStats = hitInfo.transform.GetComponent<BuddyStats>();
+		BuddyStats buddyStats = hitInfo.transform.GetComponent<BuddyStats>();
```

Comment:
It should always have data by this point, but we can add an assert to check if that's the case.